src/docs/user/ProActiveUserGuide.adoc

```diff
@@ -2611,6 +2611,103 @@ Task Synchronization API can be used for multiple dynamic synchronization patter
+The typical usage scenario of the Signal API is the following:
+
+  . A job declares to be ready to handle one or more signals. The signals are then displayed in the Workflow Execution portal (e.g., display the `soft_stop` signal)
+
+  . The job waits to receive these signals, or check for their reception regularly,
+
+  . A user clicks on the signal displayed in the Workflow Execution portal (e.g., click on the `soft_stop` signal), which sends this signal to the job,
+
+  . At the signal reception, the job executes the corresponding behavior (defined in the workflow).
+
+As already mentioned, the Signal API uses the <<_task_synchronization_api,synchronization API>> to provide a *persistent key-value channel* to store job signals.
+It uses a single channel (called *PA_SIGNALS_CHANNEL*) to store the signals of all jobs. The signal channel contains an entry per job.
+The key of each entry is the job id, whereas the value of the entry is the set of signals used by the considered job.
+Job signals are organized in a link:https://docs.oracle.com/javase/8/docs/api/java/util/Set.html[Java Set] structure,
+which means that a signal appears only once in the set, even if it is sent many times.
+All job signals are automatically removed when the job execution finishes (i.e., the job is in the status Finished, Cancelled, Failed or Killed).
+
+The Signal API is based on the <<_task_synchronization_api,synchronization API>>. However, these APIs are intended for different purposes.
+The Synchronization API is used to exchange information (e.g., messages) between tasks and between jobs, whereas the
+Signal API is useful to exchange signals between a job and users (via the Workflow Execution portal or the REST API), it cannot be used to send signals from a job to another.
+
+==== Usage
+*Task Signal API* is a *Java API* defined by the following link:../javadoc/org/ow2/proactive/scheduler/signal/SignalApi.html[API documentation].
+
+The Signal API is available inside ProActive Tasks through the *signalapi* <<_glossary_script_bindings,script binding>>.
+
+It is easy to use, and offers a set of basic operations on signals summarized as follows:
+
+  - Ready for signal(s): declares that the current job is ready to receive and handle one or more signals.
+   This operation constructs new signals (by adding the prefix `ready_` to the input signals), then adds them to job signals.
+   For example, if a job declares ready for the signal `soft_stop`, the signal `ready_soft_stop` is added to the set of job signals.
+   Additionally, if the input signal (e.g., `soft_stop`) already exists among the job signals (which means that it has been sent and handled previously),
+    the operation ready for signal automatically removes it, thus the job is ready again to receive and handle the same signal.
+
+  - Send signal(s): adds one or more signals given as input to the current job signals.
```

Comment:
when you have singular and plural functions, you can do `sendSignal`, `sendSignals` :

Reply:
OK, done